apps/src/aichat/views/ChatMessageView.tsx

```diff
@@ -0,0 +1,80 @@
+import React, {useMemo, useState} from 'react';
+
+import ChatMessage from '@cdo/apps/aiComponentLibrary/chatMessage/ChatMessage';
+import {Role} from '@cdo/apps/aiComponentLibrary/chatMessage/types';
+import {getChatMessageDisplayText} from '@cdo/apps/aiComponentLibrary/chatMessage/utils';
+import Button from '@cdo/apps/componentLibrary/button/Button';
+import {AiInteractionStatus as Status} from '@cdo/generated-scripts/sharedConstants';
+
+import aichatI18n from '../locale';
+import {ChatMessage as ChatMessageType} from '../types';
+
+import TeacherFeedbackFooter from './TeacherFeedbackFooter';
+
+import moduleStyles from '@cdo/apps/aiComponentLibrary/chatMessage/chat-message.module.scss';
+
+interface ChatMessageViewProps {
+  chatMessage: ChatMessageType;
+  isChatHistoryView: boolean;
+}
+
+const ChatMessageView: React.FunctionComponent<ChatMessageViewProps> = ({
+  chatMessage,
+  isChatHistoryView,
+}) => {
+  const [showProfaneUserMessage, setShowProfaneUserMessage] = useState(false);
+
+  const getDisplayText: string = useMemo(() => {
+    return getChatMessageDisplayText(
+      chatMessage.status,
+      chatMessage.role,
+      chatMessage.chatMessageText,
+      showProfaneUserMessage
+    );
+  }, [chatMessage, showProfaneUserMessage]);
+
+  return (
+    <ChatMessage
+      {...chatMessage}
+      showProfaneUserMessage={showProfaneUserMessage}
+    >
+      {isChatHistoryView &&
+        getDisplayText === chatMessage.chatMessageText &&
+        chatMessage.status !== Status.PROFANITY_VIOLATION && (
+          <TeacherFeedbackFooter
+            isProfanityViolation={false}
+            chatMessage={chatMessage}
+          />
+        )}
+      {isChatHistoryView &&
```

Comment:
Potentially follow-up: might be nice to break up these booleans a bit for readability (but not required)

Reply:
Do you just mean add a line break in between the 2 sections? or helper method names for clarity, or?